### PR TITLE
Updated the MTM code snippet to use IIFE

### PR DESCRIPTION
**Pull request by snake14**
### Description:

Updated the MTM code snippet to use IIFE in order to remove the declared variables from the global scope.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: docs/4.x/tagmanager/embedding.md [hunk 1]
```diff
@@ -23,10 +23,12 @@ The code looks as follows:
...
 <!-- Matomo Tag Manager -->
 <script type="text/javascript">
-    window._mtm = window._mtm || [];
-    window._mtm.push({'mtm.startTime': (new Date().getTime()), 'event': 'mtm.Start'});
+  window._mtm = window._mtm || [];
+  window._mtm.push({'mtm.startTime': (new Date().getTime()), 'event': 'mtm.Start'});
+  (function() {
     var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
-    g.type='text/javascript'; g.async=true; g.src='https://{$MATOMO_URL}/js/container_{$CONTAINER}.js'; s.parentNode.insertBefore(g,s);
+    g.type='text/javascript'; g.async=true; g.src='https://{$MATOMO_URL}/js/container_{$CONTAINER_ID}.js'; s.parentNode.insertBefore(g,s);
```
Comment (sgiehl) [inline docs/4.x/tagmanager/embedding.md:30]: You need to update the references to `{$CONTAINER}` a few lines below and somewhere near the end of that file or change it back to `{$CONTAINER}` here again. Same applies to 5.x file

File: docs/4.x/tagmanager/integration-plugin.md [hunk 2]
```diff
@@ -15,10 +15,12 @@ The easiest way to embed a container without needing any authentication is to as
...
-window._mtm = window._mtm || [];
-window._mtm.push({'mtm.startTime': (new Date().getTime()), 'event': 'mtm.Start'});
-var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
-g.type='text/javascript'; g.async=true; g.src='${MATOMOURL}/js/container_${CONTAINERID}.js'; s.parentNode.insertBefore(g,s);
+  window._mtm = window._mtm || [];
+  window._mtm.push({'mtm.startTime': (new Date().getTime()), 'event': 'mtm.Start'});
+  (function() {
+    var d=document, g=d.createElement('script'), s=d.getElementsByTagName('script')[0];
+    g.type='text/javascript'; g.async=true; g.src='https://{$MATOMO_URL}/js/container_{$CONTAINER_ID}.js'; s.parentNode.insertBefore(g,s);
+  })();
```
Comment (sgiehl) [inline docs/4.x/tagmanager/integration-plugin.md:23]: The reference to `{$CONTAINERID}` a few lines below need to be updated as well.

**Review by sgiehl — CHANGES_REQUESTED** [review]

**Review by AltamashShaikh — COMMENTED** [review]
@snake14 All looks good except Stefan's comment

File: docs/4.x/tagmanager/embedding.md (hunk 1 above)
Comment (snake14) [reply docs/4.x/tagmanager/embedding.md:30]: Thank you @sgiehl . I will make things more consistent.

File: docs/4.x/tagmanager/integration-plugin.md (hunk 2 above)
Comment (snake14) [reply docs/4.x/tagmanager/integration-plugin.md:23]: Thank you @sgiehl . I will make things more consistent.

**Review by sgiehl — APPROVED** [review]

**Pull request merged**